src/charm.py

```diff
@@ -179,9 +178,6 @@ def _on_dbcluster_relation_changed(self, event):
                     agent_id = unit_data[self.AGENT_ID_KEY]
                     all_bind_addresses[agent_id] = unit_data[self.DB_BIND_ADDR_KEY]
 
-            if self._stored.all_bind_addresses == all_bind_addresses:
-                return
-
             relation.data[self.app][self.ALL_BIND_ADDRS_KEY] = json.dumps(all_bind_addresses)
```

Comment:
We want to ensure that this is consistent every time it is written, so we should add `sort_keys=True`.

Reply:
Added